functions-and-scope.js

```diff
@@ -53,32 +92,86 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // averageGrade([6, 4, 5]) geeft xxxx
 // averageGrade([8, 9, 4, 6, 10]) geeft xxxx
 
+// let array2 = [];
+
+function averageGrade (array) {
+    const reducer = (accumulator, currentValue) => accumulator + currentValue;
+    const total = array.reduce(reducer);
+    const average = total / array.length;
+    return average;
+}
+
+console.log(averageGrade(grades));
+console.log(averageGrade([6, 4, 5]));
+console.log(averageGrade([8, 9, 4, 6, 10]));
 
 /* 2c: Afronden op twee decimalen */
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
+function averageGrade2 (array) {
+    const reducer = (accumulator, currentValue) => accumulator + currentValue;
+    const total = array.reduce(reducer);
+    const average = total / array.length;
+    const rounded = average.toFixed(2);
+    return rounded;
+}
 
-
+console.log(averageGrade2(grades));
+console.log(averageGrade2([6, 4, 5]));
+console.log(averageGrade2([8, 9, 4, 6, 10]));
 
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
 // Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe kan ik iedere waarde van de array langsgaan?
+//met een for loop itereren
 // * Op welke conditie moet ik checken?
+//of de waarde van de index hoger is dan de waarde van de ander
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+//een globale variabele aanmaken
 // Log het antwoord in de terminal.
 
 // ---- Verwachte uitkomst: 9
+let highGrade = [];
```

Comment:
De code werkt, maar als je alles in de scope zet en alleen de grades array gebruikt die buiten de scope staat dan heb je een klein stukje code dat compact bijelkaar staat.  Zou je dit ook zonder een 2de for loop kunnen maken? 
Tip: Dan werkt de 2de opdracht wel ;)